### PR TITLE
Adding fake creation options characterization tests

### PR DESCRIPTION
**Pull request by blairconrad**
As a precursor to the real work of #461, I

- moved some of the existing specs around according to the suggestion I had in https://github.com/FakeItEasy/FakeItEasy/pull/447#discussion_r24468982 and 
- added extra _characterization_ specs for the behaviour of the explicit fake creation options interactions, as laid out in https://github.com/FakeItEasy/FakeItEasy/issues/461#issuecomment-76596031. There are currently some deviations from our described ideal, which may cause us to re-evaluate that ideal:
    - WithAdditionalAttributes called twice: last one in wins, rather than stacking
    - ConfigureFake followed by Wrapping - uses configured, not wrapped, behaviour
    - CallsBaseMethods followed by Wrapping - calls base method
    - Strict followed by Wrapping - acts strict

Note that I have not changed any behaviour in the production code, only added some tests and moved a few between files.

### CODE REVIEW

File: Source/FakeItEasy.sln [hunk 1]
```diff
@@ -64,6 +64,9 @@ EndProject
 Project("{2150E333-8FDC-42A3-9474-1A3956D46DE8}") = "Integration", "Integration", "{5D41D463-D53B-4ADC-90AD-44AA03D6340D}"
 EndProject
 Project("{2150E333-8FDC-42A3-9474-1A3956D46DE8}") = "Specs", "Specs", "{575A3C3B-17BA-4A8A-9F53-05320913D0A6}"
+	ProjectSection(SolutionItems) = preProject
+		CreationOptionsSpecs.cs = CreationOptionsSpecs.cs
+	EndProjectSection
```
Comment (adamralph) [inline Source/FakeItEasy.sln]: What's this doing here? :stuck_out_tongue_closed_eyes:

Comment (adamralph) [inline Source/FakeItEasy.sln]: In fact, this looks like a no-op change. Can this whole be reverted?

**Comment by adamralph** [comment]
In the second commit, some of the test names seemed to have become less descriptive, in that they had the 'which calls a virtual xxx in constructor' removed. Is that intentional?

In the third commit, some new tests have been added which also omit that from their descriptions.

**Comment by adamralph** [comment]
Finally got round to reviewing this :wink: Sorry for taking so long and thanks for being patient.

I've finished my first pass.

File: Source/FakeItEasy.sln (hunk 1 above)
Comment (blairconrad) [inline Source/FakeItEasy.sln]: Whoa. Yes. I do not know what was going on there.
Worse, I can't explain why I didn't catch it in pre-PR review. :blush:

**Comment by blairconrad** [comment]
> Finally got round to reviewing this :wink: Sorry for taking so long and thanks for being patient.

Please, no need to apologize. Everyone's busy, and your comments are worth waiting for! :wink:

Yes, removing "which calls a virtual xxx in constructor" was intentional. I don't think it adds value. We're either talking about configuring xxxes, which is nonsensical for non-virtual xxxes, or the test class indicates that the xxxes are virtual, and the `It`s indicate when the xxxes have been called from within a constructor. (In which case I suppose the "which calls a virtual xxx in constructor" was in some sense moved, not removed.) 

If you don't care for this approach, I can reword.

If you're okay with it, I think my only action is to push a commit reverting the sln, which I can do at almost any time.

**Comment by blairconrad** [comment]
I semi-accidentally pushed the sln revert anyhow, so I figure @adamralph might as well own the PR for now. I'm still happy to rename the tests if the current names are not satisfactory.

**Comment by adamralph** [comment]
You're reasoning about the test names does indeed make sense :smile:

**Pull request merged**